zipline/pipeline/loaders/synthetic.py

```diff
@@ -346,6 +355,12 @@ def expected_bar_values_2d(dates, asset_info, colname):
             # date.
             if not (start <= date <= end):
                 continue
+            if (
```

Comment:
I tend to try to avoid in-line complex conditionals in if-statements if I can.

This might be clearer if we added an in-line `is_hole` function at the start of this function like:

```python
def is_hole(date, asset):
    return holes is not None and date in holes.get(asset, ())
```
or maybe better yet:

```python
if holes is not None:
    def is_hole(date, asset):
        return date in holes.get(asset, ())
else:
    def is_hole(date, asset):
        return False
```

Reply:
Second thought on reading this: would it make sense to push this logic down to `expected_bar_value`? That would simplify the logic in https://github.com/quantopian/zipline/pull/2303/files#diff-c2524ef7178e306daad5af35e6180d3bR388.